launch/test/launch/actions/test_group_action.py

```diff
@@ -101,3 +127,30 @@ def test_group_action_execute():
     for a in result:
         a.visit(lc1)
     assert len(lc1.launch_configurations) == 0
+
+    assert len(lc1.launch_configurations) == 0
+    lc1.launch_configurations['foo'] = 'FOO'
+    lc1.launch_configurations['bar'] = 'BAR'
+    result = GroupAction([Action()], forwarding=False,
+                         launch_configurations={'bar': LaunchConfiguration('bar'),
+                                                'baz': 'BAZ'}).visit(lc1)
+    # push, reset (which will set launch_configurations), 1 action, and pop actions
+    assert len(result) == 4
+    assert isinstance(result[0], PushLaunchConfigurations)
+    assert isinstance(result[1], ResetLaunchConfigurations)
+    assert isinstance(result[2], Action)
+    assert isinstance(result[3], PopLaunchConfigurations)
+    result[0].visit(lc1)  # Push
+    assert 'foo' in lc1.launch_configurations.keys()  # Copied to new scope, before Reset
+    assert 'bar' in lc1.launch_configurations.keys()  # Copied to new scope, before Reset
+    result[1].visit(lc1)  # Reset
+    assert 'foo' not in lc1.launch_configurations.keys()  # Cleared from scope in Reset
+    assert 'bar' in lc1.launch_configurations.keys()  # Evaluated and forwarded in Reset
```

Comment:
@djchopp nit: shall we check the launch configuration value as well?

Reply:
Added in [c7b55cf](https://github.com/ros2/launch/pull/515/commits/c7b55cfc5ea664d9d353d049beabfb7c950d0c4a)